Automattic/pocket-casts-android
# PC-I478: Include Deep Link Support for Share Links

## Description
Add deep link support for https://pca.st (for prod) and https://pcast.pocketcasts.net (for debug) share links, allowing them to be opened natively in the app and bypassing the web.

Fixes #478 

## Testing Instructions
1. Begin playing an episode
2. Open the share sheet and share either the podcast or episode
3. Paste that link into a notes app
4. Open that link
5. A system popup should appear, allowing you to open that link in the app instead of directly on the web
